lib/avro_turf.rb

```diff
@@ -43,11 +45,34 @@ def decode(encoded_data, schema_name: nil, namespace: @namespace)
     schema = schema_name && resolve_schema(schema_name, namespace)
     reader = Avro::IO::DatumReader.new(nil, schema)
     dr = Avro::DataFile::Reader.new(io, reader)
+
+    # If no reader schema is specified, we'll try to find one matching the name
+    # of the writer's schema.
+    if schema_name.nil?
+      schema = load_readers_schema(reader.writers_schema)
+      reader.readers_schema = schema if schema
+    end
+
     dr.first
   end
 
   private
 
+  # Returns the reader schema matching the writer's schema if one is locally
+  # available, or nil if there's no matching reader schema.
+  def load_readers_schema(writers_schema)
```

Comment:
Its use is pretty specific, and it logs a very specific warning. It only makes sense to load a reader schema based on an existing writer schema, not the other way around.